### ISSUE TITLE
lower_half() operation in logpdf (scipy.stats.multivariate_normal) results in asymmetric  

### ISSUE THREAD

**Issue by sanket-kamthe** [HIPS/autograd#296]
I am aware that autograd reports that logpdf is not supported, but I saw the code as I had raised an issue with it few months ago. 
However, if I just use the code in autgrad/autograd/scipy/stats/multivariate_normal.py, the resulting jacobian for covgrad is asymmetric, and gets worse for tricky derivatives. 

Reproducible simple case
```python
test_cov = np.array(([[1.0, 0.5 ], [0.5, 1.0 ]]), dtype=float)
test_mean = np.array([0.0, 0.0], dtype=float)
data = test_mean + 0.1
A = np.random.randn(2,2)
A = A @ A.T

def dummy_test(test_cov, A=A, mean=test_mean, x=data):    
    scaled_test_cov = A @ test_cov @ A.T
    return logpdf(data, test_mean, scaled_test_cov)
jacobian(dummy_test)(test_cov)

```
```sh
array([[-0.64658309, -0.613179  ],
       [ 1.18318703, -0.55036653]])
```

**Fix is easy, remove lower_half and place 0.5 in line 43** 

I can understand this may slow down everything as we are duplicating the progenitors, but just passing the lower half gets incorrect results. 

#225 still happens as the override of primitive fails sometimes and once we are inside scipy lower level function all hell breaks lose. 
PS: If you wish I can raise a separate bug to track this issue. I think it has more to do with name wrangling in Anaconda than bug in autograd itself.

**Comment by mattjj** [comment]
Thanks for the report!

There's something a bit subtle here that we've wrestled with before (e.g. #56), which is that scipy's multivariate_normal.logpdf function only reads from the lower triangle of the covariance matrix argument, and so to agree with numerical derivatives we should return an asymmetric (lower triangular) matrix. However, I think it's better to follow the convention that the logpdf function only accepts symmetric matrices in its covariance argument, and it only reading from the lower triangle is an implementation artifact (which unfortunately shows up when testing against numerical derivatives).

So I agree with your suggested fix, and that's essentially what I applied. The trickier thing was to adapt the tests, and for that I followed our convention in our test_linalg.py, which is to have special testing behavior for functions on symmetric matrices (making numerical derivatives agree with the symmetric matrix specification).

Please reopen if this failed to fix the issue!

**Comment by mattjj** [comment]
That's weird; I closed this issue from a commit that fixed it, but GitHub didn't seem to link to the commit like it usually does.

The fix is in a9a9a12, which just I pushed to master (hence the automatic closure).
